internal/handlers/telegram/handler_test.go

```diff
@@ -0,0 +1,40 @@
+package telegram
+
+import (
+	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestDocument(t *testing.T) {
+	correct := "Someone shared a file (Random File) on Telegram with caption: 'Random Caption'."
+	updateObj := &tgbotapi.Update{
+		Message: &tgbotapi.Message{
+			From: &tgbotapi.User{
+				FirstName: "Someone",
+			},
+			Document: &tgbotapi.Document{
+				FileID: "Random File",
+			},
+			Caption: "Random Caption",
+		},
+	}
+	// Call the String() function which checks if a username(Optional) exists
+	updateObj.Message.From.UserName = updateObj.Message.From.String()
+
+	// Assign the MimeType(Optional) to FileID(Required) if it does not exists
+	if updateObj.Message.Document.MimeType == ""{
+		updateObj.Message.Document.MimeType = updateObj.Message.Document.FileID
+	}
+
+	// Change to FileName(Optional) is the Caption(Optional) is not given
+	if updateObj.Message.Caption == ""{
+		updateObj.Message.Document.FileName = "Random Text"
+	}
```

Comment:
I am cautious about Caption and FileName. They are both optional, meaning that a document may not have either a caption or a filename attached. The test might fail in that case. These are the two options we used in the `handler.go` file. Should we add another option or remove both Caption and FileName from how we format the string sent to IRC?

Reply:
I feel the caption is very useful to users on IRC. Even if there was no caption set, I feel that is still useful information to IRC users.